src/android/StatusBar.java

```diff
@@ -210,6 +229,17 @@ private void setStatusBarBackgroundColor(final String colorPref) {
         if (Build.VERSION.SDK_INT >= 21) {
             if (colorPref != null && !colorPref.isEmpty()) {
                 final Window window = cordova.getActivity().getWindow();
+
+                // This is so that we can return to the last set statusbar colours if overlay state changes.
+                int uiOptions = window.getDecorView().getSystemUiVisibility();
+                boolean isOverlayed = ((uiOptions | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN) == uiOptions);
```

Comment:
Do we need to take `SYSTEM_UI_FLAG_LAYOUT_STABLE` into account, too, or is `SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN` enough?

Reply:
I think that `SYSTEM_UI_FLAG_LAYOUT_STABLE` will help a lot. In our code, we've delayed the rendering of the map for a harcoded time of 160ms (10 frames at 60fps), just to get around this issue. If we don't add that delay, the map will be resized immediately after it finishes rendering for the first time, and it doesn't look good.

Reply:
And I think that this method should have a different name, It's name is a bit misleading; You can see that it's used for the `overlaysWebView` action, but it's name doesn't suggest anything on those lines.
```
        if ("overlaysWebView".equals(action)) {
            // ...
        }
```

Reply:
I'll try the [SYSTEM_UI_FLAG_LAYOUT_STABLE](https://developer.android.com/reference/android/view/View#SYSTEM_UI_FLAG_LAYOUT_STABLE), however it is now deprecated... along with the other system ui flag constants. But it looks like the replacement is only going to be available in the API 30 (unreleased, under API R), soooo we will have no choice but to continue using them for the next foreseeable future.

> In our code, we've delayed the rendering of the map for a harcoded time of 160ms (10 frames at 60fps), just to get around this issue. If we don't add that delay, the map will be resized immediately after it finishes rendering for the first time, and it doesn't look good.

Makes me wonder if this will correct https://github.com/apache/cordova-plugin-statusbar/issues/158 ...